server/gameconnection.py

```diff
@@ -294,14 +295,19 @@ async def handle_game_result(self, army, result):
         except (KeyError, ValueError):  # pragma: no cover
             self._logger.warning("Invalid result for %s reported: %s", army, result)
 
-    async def handle_operation_complete(self, army, secondary, delta):
-        if not int(army) == 1:
+    async def handle_operation_complete(
+        self,
+        primary_objectives_complete: typing.Union[bool, int],
+        secondary_objectives_complete: typing.Union[bool, int],
+        delta: str,
+    ) -> None:
+        if not primary_objectives_complete:
```

Comment:
the var is actually a string, so this will only reject `""` which is not what we want.
```suggestion
        if not bool(primary_objectives_complete):
```

Reply:
`bool` seems like a bad idea if we can get boolean strings too?

```python
>>> bool("false") is True
True
```

Reply:
You could do something similar to this, or maybe just use this class: 
https://github.com/FAForever/server/blob/aa535a92b676525f67223be8035fb132ca106f60/server/games/typedefs.py#L214-L218